virtcontainers/acrn_arch_base.go

```diff
@@ -0,0 +1,762 @@
+// Copyright (c) 2019 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package virtcontainers
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/kata-containers/runtime/virtcontainers/device/config"
+	"github.com/kata-containers/runtime/virtcontainers/types"
+	"github.com/sirupsen/logrus"
+)
+
+type acrnArch interface {
+
+	// acrnPath returns the path to the ACRN binary
+	acrnPath() (string, error)
+
+	// acrnctlPath returns the path to the ACRNCTL binary
+	acrnctlPath() (string, error)
+
+	// kernelParameters returns the kernel parameters
+	// if debug is true then kernel debug parameters are included
+	kernelParameters(debug bool) []Param
+
+	//capabilities returns the capabilities supported by ACRN
+	capabilities() types.Capabilities
+
+	// memoryTopology returns the memory topology using the given amount of memoryMb and hostMemoryMb
+	memoryTopology(memMb uint64) Memory
+
+	// appendConsole appends a console to devices
+	appendConsole(devices []Device, path string) []Device
+
+	// appendImage appends an image to devices
+	appendImage(devices []Device, path string) ([]Device, error)
+
+	// appendBridges appends bridges to devices
+	appendBridges(devices []Device) []Device
+
+	// appendLPC appends LPC to devices
+	// UART device emulated by the ACRN-DM is connected to the system by the LPC bus
+	appendLPC(devices []Device) []Device
+
+	// appendSocket appends a socket to devices
+	appendSocket(devices []Device, socket types.Socket) []Device
+
+	// appendNetwork appends a endpoint device to devices
+	appendNetwork(devices []Device, endpoint Endpoint) []Device
+
+	// appendBlockDevice appends a block drive to devices
+	appendBlockDevice(devices []Device, drive config.BlockDrive) []Device
+
+	// handleImagePath handles the Hypervisor Config image path
+	handleImagePath(config HypervisorConfig)
+}
+
+type acrnArchBase struct {
+	path                 string
+	ctlpath              string
+	kernelParamsNonDebug []Param
+	kernelParamsDebug    []Param
+	kernelParams         []Param
+}
+
+const acrnPath = "/usr/bin/acrn-dm"
+const acrnctlPath = "/usr/bin/acrnctl"
```

Comment:
If wonder if these paths are going to be reliable.

Reply:
from the Makefile, I see the path for other hypervisors are in /usr/bin. How does kata handle this for qemu?